lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -150,17 +150,33 @@ def method_completion_candidates(name, receiver_name)
       ancestors = linearized_ancestors_of(receiver_name)
 
       candidates = name ? prefix_search(name).flatten : @entries.values.flatten
-      candidates.filter_map do |entry|
+      completion_items = candidates.each_with_object({}) do |entry, hash|
         case entry
         when Entry::Member, Entry::MethodAlias
-          entry if ancestors.any?(entry.owner&.name)
+          entry_name = entry.name
+          ancestor_index = ancestors.index(entry.owner&.name)
+          existing_entry = hash[entry_name]
+
+          # If this method belongs to an ancestors and either we haven't found it yet or it appears earlier in the
+          # ancestors chain than the previous candidate, then we need to put it in the hash. Otherwise, continue
```

Comment:
Could you explain this part with an example? I don't quite get it 🤔

Reply:
I tried expanding the explanation. Let me know if it's clear or confusing.